### PR TITLE
Static Analysis: forgotten return statements

### PR DESCRIPTION
**Pull request by soad003**
As described in: https://github.com/ethereum/solidity/issues/3026#issuecomment-334399052

### CODE REVIEW

**Comment by soad003** [comment]
I don't know why the browser test fails. Can not reproduce it. Did not get the test to run locally...

**Comment by axic** [comment]
@yann300 do you have any idea why the test could fail?

File: src/app/staticanalysis/modules/staticAnalysisCommon.js [hunk 1]
```diff
@@ -257,6 +258,18 @@ function getFunctionOrModifierDefinitionParameterPart (funcNode) {
...
 
 /**
+ * Returns return parameter node for a function or modifier definition, Throws on wrong node.
+ * Example:
+ * function bar(uint a, uint b) returns (bool a, bool b) => bool a, bool b
+ * @funcNode {ASTNode} Contract Definition node
+ * @return {parameterlist node} parameterlist node
+ */
+function getFunctionOrModifierDefinitionReturnParameterPart (funcNode) {
+  if (!isFunctionDefinition(funcNode) && !isModifierDefinition(funcNode)) throw new Error('staticAnalysisCommon.js: not an function definition')
```
Comment (axic) [inline src/app/staticanalysis/modules/staticAnalysisCommon.js]: `not an` -> `not a`

File: src/app/staticanalysis/modules/noReturn.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  return func.returns.length > 0
+}
+
+function hasReturnStatement (func) {
+  return func.relevantNodes.filter(common.isReturn).length > 0
+}
+
+function hasAssignToAllNamedReturns (func) {
+  var namedReturns = func.returns.filter((n) => n.name.length > 0).map((n) => n.name)
+  var assignedvars = func.relevantNodes.filter(common.isAssignment).map(common.getEffectedVariableName)
```
Comment (axic) [inline src/app/staticanalysis/modules/noReturn.js]: Most of this code uses camelcase, perhaps this one should be camelcase (`assignedVars`) too.

**Comment by axic** [comment]
I am not familiar with the static analyzer but the changes look reasonable.

**Comment by axic** [comment]
@chriseth can you review this please?

**Comment by yann300** [comment]
failing tests are a mistery sometimes... I think rebasing would help . could you do that?

**Comment by soad003** [comment]
@yann300: did the rebase. still the tests fail. Could you check if they run on your machine?

**Comment by axic** [comment]
@soad003 @yann300 this now seems to pass the tests?

**Comment by yann300** [comment]
yes I think we can merge now, I will just add browser-test right after the merge cause static analysis report formatting changed a bit (it should return a yoyo element and not a string)

**Comment by axic** [comment]
@yann300 merged this now since I wasn't sure when/how you would want to add those tests. Sorry if I've messed up something with it.

**Pull request merged**